_source/_posts/2022-06-19-kubernetes-jhipster-aws.md

```diff
@@ -0,0 +1,778 @@
+  Blog{user(login)} to User
+  Post{blog(name)} to Blog
+}
+
+relationship ManyToMany {
+  Post{tag(name)} to Tag{post}
+}
+
+paginate Post, Tag with infinite-scroll
+paginate Product with pagination
+
+microservice Product with store
+microservice Blog, Post, Tag with blog
+
+deployment {
+  deploymentType kubernetes
+  appsFolders [gateway, blog, store]
+  dockerRepositoryName "{your-docker-repository-name}"
+  kubernetesNamespace demo
+}
+```
+
+From the root project directory, run the generator.
+
+```bash
+jhipster jdl app.jdl
+```
+
+This will probably take a minute or two to complete.
+
+It should end with something like the following output.
+
+```bash
+...
+
+INFO! Alternatively, you can use Jib to build and push image directly to a remote registry:
+  ./gradlew bootJar -Pprod jib -Djib.to.image={your-docker-repository-name}/gateway in /home/andrewcarterhughes/Development/okta/2022/aws-kubernetes/gateway
+  ./gradlew bootJar -Pprod jib -Djib.to.image={your-docker-repository-name}/blog in /home/andrewcarterhughes/Development/okta/2022/aws-kubernetes/blog
+  ./gradlew bootJar -Pprod jib -Djib.to.image={your-docker-repository-name}/store in /home/andrewcarterhughes/Development/okta/2022/aws-kubernetes/store
+
+You can deploy all your apps by running the following kubectl command:
+  bash kubectl-apply.sh -f
+
+[OR]
+
+If you want to use kustomize configuration, then run the following command:
+  bash kubectl-apply.sh -k
+
+Use these commands to find your application's IP addresses:
+  kubectl get svc gateway -n demo
+
+INFO! Generator kubernetes succeed
+Congratulations, JHipster execution is complete!
+Sponsored with ❤️  by @oktadev.
+```
+
+If you look at the directories in your root project, you'll see four subdirectories.
+
+```bash
+├── blog
+├── gateway
+├── kubernetes
+└── store
+```
+
+Three of the directories are the application projects for your services: blog, store, and gateway. Remember that the registry uses a stock image, so it doesn't need a local project. The fourth directory, kubernetes, holds the descriptor files for the kubernetes deployements (YAML files that tell Kubernetes how to deploy your applications from the Docker images).
+
+If you looked in that directory, you'd see four subdirectories (as well as some other files).
+
+```bash
+├── blog-k8s
+├── gateway-k8s
+├── registry-k8s
+└── store-k8s
+```
+
+These four directories hold the deployment descriptors for the Kubernetes services that correspond to the four applications in the microservice.
+
+## Fix Eureka OUT_OF_SERVICE Error
+
+There is an issue with Eureka that currently requires a workaround. You can read about the issue on [Spring Cloud Netflix](https://github.com/spring-cloud/spring-cloud-netflix/issues/3941) and [Netflix Eureka](https://github.com/Netflix/eureka/issues/1398). Essentially what happens is that the app gets stuck as `OUT_OF_SERVICE` and Eureka fails to register when it comes back up. This only happens with the store in this microservice.
```

Comment:
Thats strange, I didn't have such an issue on EKS with JHipster Registry (I did something similar last week on EKS for a [blog](https://github.com/oktadev/okta-blog/pull/1189)), may be the issue was present but the apps and registry restarted few times and got the right status as expected in k8s. Did you see if waiting for an extra minute or two worked?

Reply:
I can try again. I had to do this on Azure.

Reply:
You are right. This is unnecessary.